app/src/main/java/com/willowtree/vocable/MainActivity.kt

```diff
@@ -10,129 +11,166 @@ import android.view.Surface
 import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ViewModelOwner
+import org.koin.androidx.viewmodel.ext.android.viewModel
 import timber.log.Timber
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(),
+                     EasyPermissions.PermissionCallbacks {
 
     private val minOpenGlVersion = 3.0
     private val displayMetrics = DisplayMetrics()
     private var currentView: View? = null
-    private lateinit var viewModel: FaceTrackingViewModel
     private var paused = false
     private lateinit var binding: ActivityMainBinding
     private val sharedPrefs: VocableSharedPreferences by inject()
+    private var hasSetupAr: Boolean = false
     private val allViews = mutableListOf<View>()
-    private val settingsViewModel: SettingsViewModel by viewModels()
+
+    private val selectionModeViewModel: SelectionModeViewModel by viewModel(owner = {
+        ViewModelOwner.from(this)
+    })
+    private val faceTrackingViewModel: FaceTrackingViewModel by viewModel(owner = {
+        ViewModelOwner.from(this)
+    })
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        binding.pointerView.isVisible = false
+        setContentView(binding.root)
+
         val shouldForceDisableHeadTracking = !BuildConfig.USE_HEAD_TRACKING
         val isNotSupportedDevice = !checkIsSupportedDeviceOrFinish()
 
-        if (shouldForceDisableHeadTracking || isNotSupportedDevice) {
-            return
+        if (!shouldForceDisableHeadTracking && !isNotSupportedDevice) {
+            selectionModeViewModel.headTrackingPermissionState.observe(this) { headTrackingState ->
+                when (headTrackingState) {
+                    HeadTrackingPermissionState.PermissionRequested -> requestPermissions()
+                    HeadTrackingPermissionState.Enabled -> {
+                        togglePointerVisible(true)
+                        setupArTracking()
+                    }
+
+                    HeadTrackingPermissionState.Disabled -> {
+                        togglePointerVisible(false)
+                    }
+                }
+            }
         }
 
-        if (supportFragmentManager.findFragmentById(R.id.face_fragment) == null && BuildConfig.USE_HEAD_TRACKING) {
-            supportFragmentManager
-                .beginTransaction()
-                .replace(R.id.face_fragment, FaceTrackFragment())
-                .commitAllowingStateLoss()
-        } else {
-            window
-                .decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                .or(View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION)
-                .or(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN)
-                .or(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION)
-                .or(View.SYSTEM_UI_FLAG_FULLSCREEN)
-                .or(View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
```

Comment:
I didn't notice them doing anything. If anything I think they are unneeded as they move to a `statusBar` removed full screen state. Something that looked copied from `ArFragment`. Since we are not a traditional AR experience it seems unnecessary.

Reply:
> If anything I think they are unneeded as they move to a statusBar removed full screen state

But is changing that behavior related to intuitively asking for Camera permission? This behavior is right on the line of what I'd consider more of a "product change", so at the very least, it should exist in its own PR with a ticket linking it. Then if the change is contested later on, there's a paper trail of where this change came from and why. If someone sees this was removed, and the commit history shows it was removed as part of a ticket related to permissions, that will be confusing.